adafruit_minimqtt/adafruit_minimqtt.py

```diff
@@ -985,15 +956,27 @@ def _sock_exact_recv(self, bufsize):
         bytes is returned or trigger a timeout exception.
 
         :param int bufsize: number of bytes to receive
-
+        :return: byte array
         """
+        stamp = time.monotonic()
         if not self._backwards_compatible_sock:
             # CPython/Socketpool Impl.
             rc = bytearray(bufsize)
-            self._sock.recv_into(rc, bufsize)
-        else:  # ESP32SPI Impl.
-            stamp = time.monotonic()
+            mv = memoryview(rc)
+            recv_len = self._sock.recv_into(rc, bufsize)
+            to_read = bufsize - recv_len
+            assert to_read >= 0
```

Comment:
Want to raise a descriptive exception here, instead?

Reply:
Sure, it will provide more data that can be used for debugging.